### PR TITLE
Mapped some stuff about fish, especially fish group and tropical fish

### PR DESCRIPTION
**Pull request by liach**
also include some mob spawning/despawning mappings

Signed-off-by: liach <liach@users.noreply.github.com>

### CODE REVIEW

File: mappings/net/minecraft/entity/passive/GroupingFishEntity.mapping [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+		ARG 1 type
+	METHOD N getLimitPerChunk ()I
+	METHOD W_ update ()V
+	METHOD a joinGroupOf (Lalz;)Lalz;
+	METHOD a (Lbbq;Lafn;Lahh;Lahp;Lhs;)Lahp;
+	METHOD a pullInOtherFish (Ljava/util/stream/Stream;)V
+	METHOD dB getMaxGroupSize ()I
+	METHOD dC hasLeader ()Z
+	METHOD dD leaveGroup ()V
+	METHOD dE canHaveMoreFishesInGroup ()Z
```
Comment (asiekierka) [inline mappings/net/minecraft/entity/passive/GroupingFishEntity.mapping]: Fishes? :thinking:

**Review by asiekierka — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/entity/passive/GroupingFishEntity.mapping [hunk 2]
```diff
@@ -0,0 +1,23 @@
+CLASS alz net/minecraft/entity/passive/GroupingFishEntity
```
Comment (Gegy) [inline mappings/net/minecraft/entity/passive/GroupingFishEntity.mapping]: I think `school` would be more correct within the context of fish.

File: mappings/net/minecraft/entity/passive/GroupingFishEntity.mapping (hunk 1 above)
Comment (sciguyryan) [reply mappings/net/minecraft/entity/passive/GroupingFishEntity.mapping]: https://grammarist.com/usage/fish-fishes/

Apparently it's valid as long as it's multiple species of fish. Who knew?

File: mappings/net/minecraft/entity/mob/MobEntity.mapping [hunk 3]
```diff
@@ -27,7 +27,7 @@ CLASS ahf net/minecraft/entity/mob/MobEntity
 	METHOD A playAmbientSound ()V
 	METHOD D getAmbientSound ()Lxm;
 	METHOD F getLootTableId ()Lqc;
-	METHOD G ()Z
+	METHOD G cannotDespawn ()Z
```
Comment (Prospector) [inline mappings/net/minecraft/entity/mob/MobEntity.mapping:30]: Does this get if it cannot despawn or set it so it cannot despawn?

File: mappings/net/minecraft/entity/passive/GroupingFishEntity.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/entity/passive/GroupingFishEntity.mapping]: Fixed.

File: mappings/net/minecraft/entity/passive/GroupingFishEntity.mapping (hunk 2 above)
Comment (liach) [reply mappings/net/minecraft/entity/passive/GroupingFishEntity.mapping]: Done.

File: mappings/net/minecraft/entity/mob/MobEntity.mapping (hunk 3 above)
Comment (liach) [reply mappings/net/minecraft/entity/mob/MobEntity.mapping:30]: You definitely know this is a getter. A setter should be named `setNoDespawn` here instead.
`canXxx` `isXxx` are regular getter method name patterns, while `setXxx` is the setter.
`canXxx` is a verb phrase, so it is a good method name.

**Comment by liach** [comment]
@asiekierka Can you review again?

File: mappings/net/minecraft/entity/passive/GroupingFishEntity.mapping (hunk 2 above)
Comment (Gegy) [reply mappings/net/minecraft/entity/passive/GroupingFishEntity.mapping]: Maybe should apply to other contexts where `group` is used?

**Pull request merged**